### PR TITLE
dep(global): update kotlin, spring, spring-boot and spring-security versions

### PR DESCRIPTION
**Pull request by Pittinic**
Related to:
- https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-2.6-Release-Notes#pathpattern-based-path-matching-strategy-for-spring-mvc
- https://spring.io/blog/2022/02/21/spring-security-without-the-websecurityconfigureradapter
- https://springdoc.org/#migrating-from-springfox

Done:
- update dependencies
- remove deprecated usages
    + Experimental annotation
    + WebSecurityConfigurerAdapter replaced by DSLs such as SecurityFilterChain in a basic @Configuration class
    + HandlerInterceptorAdapter abstract class replaced by HandlerInterceptor interface
- add mysql demo module to offer a closer-to-business-case yet simple testing environment
- update version from 0.1.4-RC to 0.2.0-RC

ToDo (priorities are to-be-discussed and to-be-defined):
- update com.gnirps.keycloak.annotations.UnsecuredAccess's RequiresOptIn annotation's message
- migrate from springfox to springdoc
- review error handlers in all modules (to factorise basic behaviour and make specific handlers catch specific errors only - to let anything unrelated be caught by another handler with a lesser priority level)
- upgrade build scripts and docker config

### CODE REVIEW

**Review by bendevneuroo — APPROVED** [review]

File: gnirps-origin/pom.xml [hunk 1]
```diff
@@ -178,18 +179,31 @@
                 <artifactId>acme4j-utils</artifactId>
                 <version>${acme4j.version}</version>
             </dependency>
+
+            <!-- bcprov-jdk15on -->
             <dependency>
-                <groupId>commons-io</groupId>
-                <artifactId>commons-io</artifactId>
-                <version>${commons-io.version}</version>
+                <groupId>org.bouncycastle</groupId>
```
Comment (vainj) [inline gnirps-origin/pom.xml:185]: question(non-blocking): Y-aurait-il un intérêt à ajouter un commentaire pour expliquer pourquoi certaines dépendances ont été ajoutées dans ce `pom.xml` ?

File: keycloak/src/main/kotlin/com/gnirps/keycloak/config/KeycloakConfigurerAdapter.kt [hunk 2]
```diff
@@ -61,40 +60,13 @@ class KeycloakConfigurerAdapter(
     }
 
     /**
-     * Ignore authentication for some urls at the "Spring Security" level
-     * @param web
+     * Spring-security déconseille d'utiliser le WebSecurity.ignoring pour échapper la sécurité sur certains URL, car
+     * cela ne les fait pas passer dans Spring-security. Il vaut mieux utiliser HttpSecurity.anyMatcher().permitAll()
+     * qui est un laissez-passer a posteriori et donc passe quand même dans les Spring-security
```
Comment (vainj) [inline keycloak/src/main/kotlin/com/gnirps/keycloak/config/KeycloakConfigurerAdapter.kt:65]: todo(non-blocking): Par soucis de cohérence avec le reste de l'application, pourrions-nous traduire ce commentaire en Anglais ?

File: gnirps-origin/pom.xml [hunk 3]
```diff
@@ -178,18 +179,31 @@
...
+                <artifactId>bcprov-jdk15on</artifactId>
+                <version>1.70</version>
```
[the remark this answers is not in the document]
Comment (bendevneuroo) [reply gnirps-origin/pom.xml:187]: done

File: keycloak/src/main/kotlin/com/gnirps/keycloak/config/KeycloakConfigurerAdapter.kt (hunk 2 above)
Comment (bendevneuroo) [reply keycloak/src/main/kotlin/com/gnirps/keycloak/config/KeycloakConfigurerAdapter.kt:65]: pris en compte

File: gnirps-origin/pom.xml (hunk 1 above)
Comment (bendevneuroo) [reply gnirps-origin/pom.xml:198]: pris en compte 756e6c93d2aa0f712d86f6b2c1a077c6f29835f9

**Review by vainj — APPROVED** [review]

**Pull request merged**